packages/react-components/react-jsx-runtime/src/createElement.ts

```diff
@@ -10,40 +10,29 @@ export function createElement<P extends {}>(
   props?: P | null,
   ...children: React.ReactNode[]
 ): React.ReactElement<P> | null {
-  if (!isSlotComponent(props)) {
-    return React.createElement(type, props, ...children);
+  return isSlotComponent(props)
+    ? createElementFromSlot(type, props, children)
+    : React.createElement(type, props, ...children);
+}
+
+function createElementFromSlot<P extends UnknownSlotProps>(
```

Comment:
```suggestion
function createElementFromRenderFunction<P extends UnknownSlotProps>(
```

This function does not handle slots, it handles only a specific scenario with slots. Is it more accurate? (feel to keep as it is/propose better name)

Reply:
Sounds good to me, I guess I kept the name as `isSlotComponent` thinking on the future implementation